src/SqlAsyncEnumerable.cs

```diff
@@ -105,7 +108,10 @@ private async Task<bool> GetNextRowAsync()
                 {
                     using (SqlCommand command = SqlBindingUtilities.BuildCommand(this._attribute, this._connection))
                     {
-                        await command.Connection.OpenAsync();
+                        if (this._connection.State != System.Data.ConnectionState.Open)
+                        {
+                            await this._connection.OpenAsync();
```

Comment:
Is there a reason you're doing this here? The connection shouldn't be closed so I wouldn't expect this to be necessary.

Reply:
If we did want to add that we should add it every time we read a value - since at that point we're saying that we expect the connection could be closed at any point. But if we did that let's do it as a separate task and just keep the behavior as it is now, there's no reasonable reason I can think of for why the connection would have been closed between the creation of the enumerable and querying the rows.